source/Nevermore/Transient/DbCommandExtensions.cs

```diff
@@ -128,39 +84,96 @@ public static async Task<object> ExecuteScalarWithRetryAsync(this DbCommand comm
             var effectiveCommandRetryPolicy = (commandRetryPolicy ?? RetryManager.Instance.GetDefaultSqlCommandRetryPolicy()).LoggingRetries(operationName);
             return await effectiveCommandRetryPolicy.ExecuteActionAsync(async () =>
             {
-                var weOwnTheConnectionLifetime = EnsureValidConnection(command, connectionRetryPolicy);
-                try
-                {
-                    return await command.ExecuteScalarAsync(cancellationToken);
-                }
-                finally
-                {
-                    if (weOwnTheConnectionLifetime && command.Connection != null && command.Connection.State == ConnectionState.Open)
-                        await command.Connection.CloseAsync();
-                }
+                await using var validCommand = await ValidCommandAsync.For(command, connectionRetryPolicy, cancellationToken);
+                return await validCommand.ExecuteScalarAsync(cancellationToken);
             });
         }
 
         static void GuardConnectionIsNotNull(DbCommand command)
         {
-            if (command.Connection == null)
+            if (command?.Connection == null)
```

Comment:
This is an interesting one... when this code was used as part of the EnsureValidConnection call, it verified that command was not null, then skipped it if it was... but ALL the methods call this Guard Method first... and if command was null it would have thrown a null exception error anyway.... so there is a slight difference here, in that a null command will now throw a connection property exception... should I revert this?